File: Week 01/Seminar/TK/single_number/solution_sort.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution
{
public:
    int singleNumber(vector<int> &nums)
    {
        sort(nums.begin(), nums.end());

        for (int i = 0; i < nums.size(); i += 2)
        {
            if (i == nums.size() - 1 || nums[i] != nums[i + 1])
            {
                return nums[i];
            }
        }

        return 0;
    }
};

File: Week_08/Seminar/Solutions/longest_consecutive_sequence/solution_sort.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0 

        nums.sort()
        max_count = count = 1

        for i in range(len(nums) - 1):
            if nums[i] == nums[i + 1]:
                continue
                
            if nums[i] + 1 == nums[i + 1]:
                count += 1
                max_count = max(count, max_count)
            else:
                count = 1
        
        return max_count
        